VocaDbWeb.Core/Controllers/StatsController.cs

```diff
@@ -34,27 +35,20 @@ public class StatsController : ControllerBase
 		private readonly ActivityEntryQueries _activityEntryQueries;
 		private readonly StatsQueries _queries;
 		private readonly SongAggregateQueries _songAggregateQueries;
+		private readonly ObjectCache _cache;
 
-		// TODO: implement
-		/*private T GetCachedReport<T>() where T : class
+		private T GetCachedReport<T>() where T : class
 		{
 			var name = ControllerContext.RouteData.Values["action"] + "_" + ControllerContext.RouteData.Values["cutoff"];
-			var item = _context.Cache["report_" + name];
-
-			if (item == null)
-				return null;
-
-			return (T)item;
-		}*/
-		private T GetCachedReport<T>() where T : class => throw new NotImplementedException();
+			var item = _cache.Get($"report_{name}");
+			return item as T;
+		}
 
-		// TODO: implement
-		/*private void SaveCachedReport<T>(T data) where T : class
+		private void SaveCachedReport<T>(T data) where T : class
 		{
 			var name = ControllerContext.RouteData.Values["action"];

```

Comment:
Could you move this to property, otherwise it looks fine

Reply:
Caching doesn't seem to work since the cache keys don't match? I replaced `GetCachedReport` and `SaveCachedReport` with `GetOrInsert` instead of moving this to a property.

Reply:
Caching doesn't seem to work properly since the cache keys don't match? I replaced `GetCachedReport` and `SaveCachedReport` with `GetOrInsert` instead of moving this to a property.